File: python/list-all-project-keywords.py
#
# GitHub: https://github.com/SERPWoo/API
#
# This code requests a project's keywords and outputs the Keyword ID, Keyword, PPC Comp, OCI, Volume, CPC (USD), created date, oldest date, recent date, Link To SERPs
#
# This output is text format
#
# Last updated - Aug 24th, 2017 @ 22:22 EST (@MercenaryCarter https://github.com/MercenaryCarter and https://twitter.com/MercenaryCarter)
#
# Run Command: python list-all-project-keywords.py

import requests, json

def main():        
    API_key = 'API_KEY_HERE'
    Project_ID = 0 #input your Project ID

    url = "https://api.serpwoo.com/v1/projects/" + str(Project_ID) + "/keywords/"
    payload = {'key': API_key}

    r = requests.get(url, params=payload)
    j = r.json()

    print('\n--\n')

    print('%-15s %-50s %-10s %-10s %-15s %-10s %-15s %-15s %-15s %-40s' % ("Keyword ID", "Keyword", "PPC Comp", "OCI", "Search Volume", "CPC (USD)", "Created Date", "Oldest Date", "Recent Date", "Link To SERPs"))
    print('%-15s %-50s %-10s %-10s %-15s %-10s %-15s %-15s %-15s %-40s' % ("----------", "-------", "--------", "---", "-------------", "---------", "------------", "-----------", "-----------", "-------------"))

    if j['success'] == 1:
        for project_id in j['projects']:

            for keyword_id in sorted(j['projects'][project_id]['keywords'], cmp=_compare_keys):

                print('%-15s %-50s %-10s %-10s %-15s %-10s %-15s %-15s %-15s %-40s' % (keyword_id, j['projects'][project_id]['keywords'][keyword_id]['keyword'], j['projects'][project_id]['keywords'][keyword_id]['Comp'], j['projects'][project_id]['keywords'][keyword_id]['oci'], j['projects'][project_id]['keywords'][keyword_id]['volume'], j['projects'][project_id]['keywords'][keyword_id]['CPC']['usd']['amount'], j['projects'][project_id]['keywords'][keyword_id]['creation_date'], j['projects'][project_id]['keywords'][keyword_id]['SERP_data']['oldest_date'], j['projects'][project_id]['keywords'][keyword_id]['SERP_data']['recent_date'], j['projects'][project_id]['keywords'][keyword_id]['_links']['serps']))

    else:

        print('An Error Occurred: ' + j['error'])


    print('\n--\n')


#Sorting comparion
def _compare_keys(x, y):
    try:
        x = int(x)
    except ValueError:
        xint = False
    else:
        xint = True
    try:
        y = int(y)
    except ValueError:
        if xint:
            return -1
        return cmp(x.lower(), y.lower())
        # or cmp(x, y) if you want case sensitivity.
    else:
        if xint:
            return cmp(x, y)
        return 1


################################################################################
main()

File: go/list-all-project-keywords.go
//
// Github Repo: https://github.com/SERPWoo/API
//
// This code requests a project's keywords and outputs the Keyword ID, Keyword, PPC Comp, OCI, Volume, CPC (USD), created date, oldest date, recent date, Link To SERPs
//
// This output is text format
//
// Last updated - Aug 26th, 2017 @ 16:35 EST (@MercenaryCarter https://github.com/MercenaryCarter and https://twitter.com/MercenaryCarter)
//
// Run Command: go run list-all-project-keywords.go
//

		package main

		import (
			"net/http"
			"log"
			"encoding/json"
			"fmt"
			"errors"
			"sort"
			"strconv"
		)

		//for the Keyword Structure
		type KeywordData struct {
			Keyword			string	`json:"keyword"`
			Comp			string	`json:"Comp"`
			OCI				string	`json:"oci"`
			Volume			int		`json:"volume"`
			Creation_Date	int		`json:"creation_date"`

			Links		struct {
							SERPs	string		`json:"serps"`
						} `json:"_links"`

			SDATA		struct {
							Recent		int		`json:"recent_date"`
							Oldest		int		`json:"oldest_date"`
						} `json:"SERP_data"`

			CPC		struct {
						USD		struct {
									Amount	string		`json:"amount"`
								} `json:"usd"`								
					} `json:"CPC"`

		}

func main() {

	// Get your API Key here: https://www.serpwoo.com/v3/api/ (should be logged in)
	var API_key = "API_KEY_HERE"
	var Project_ID = 0 //input your Project ID

	JSONoutput, err := Call_Project_Keywords_API(strconv.Itoa(Project_ID), API_key)

		if err != nil {
			log.Fatal(err)
		}

	fmt.Printf("\n--\n")

	var keyword_ids []int

		for k, _ := range JSONoutput {
			keyword_ids = append(keyword_ids, k)
		}

	sort.Sort(sort.IntSlice(keyword_ids))

		if len(keyword_ids) > 0 {
			fmt.Printf("%-15s %-50s %-10s %-10s %-15s %-10s %-15s %-15s %-15s %-40s\n", "Keyword ID", "Keyword", "PPC Comp", "OCI", "Search Volume", "CPC (USD)", "Created Date", "Oldest Date", "Recent Date", "Link To SERPs")
			fmt.Printf("%-15s %-50s %-10s %-10s %-15s %-10s %-15s %-15s %-15s %-40s\n", "----------", "-------", "--------", "---", "-------------", "---------", "------------", "-----------", "-----------", "-------------")
		}

		for _, keyword_id := range keyword_ids {
			fmt.Printf("%-15d %-50s %-10s %-10s %-15d %-10s %-15d %-15d %-15d %-40s\n", keyword_id, JSONoutput[keyword_id].Keyword, JSONoutput[keyword_id].Comp, JSONoutput[keyword_id].OCI, JSONoutput[keyword_id].Volume, JSONoutput[keyword_id].CPC.USD.Amount, JSONoutput[keyword_id].Creation_Date, JSONoutput[keyword_id].SDATA.Oldest, JSONoutput[keyword_id].SDATA.Recent, JSONoutput[keyword_id].Links.SERPs)
		}

	fmt.Printf("\n--\n")

}


////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////


func Call_Project_Keywords_API(pid, api_key string) (map[int]KeywordData, error) {

	var printer map[int]KeywordData

	url := "https://api.serpwoo.com/v1/projects/" + pid + "/keywords/?key=" + api_key

	out, err := http.Get(url)

		if err != nil {
			return printer, err
		}

	var JSONoutput interface{}
	defer out.Body.Close()

	err = json.NewDecoder(out.Body).Decode(&JSONoutput)
	data, ok := JSONoutput.(map[string]interface{})["projects"].(map[string]interface{})[pid]

		if !ok {
			return printer, errors.New("Error: Project ID " + pid + " was not found.")
		}

	final, ok := data.(map[string]interface{})["keywords"].(map[string]interface{})

		if !ok {
			return printer, errors.New("Error: Unable to process keywords within the tag struct.")
		}

	bt, err := json.Marshal(final)

		if err != nil {
			return printer, err
		}

	err = json.Unmarshal(bt, &printer)

		if err != nil {
			return printer, err
		}

	return printer, nil
}